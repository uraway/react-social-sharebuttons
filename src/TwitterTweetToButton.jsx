// @flow
import React, { Component } from 'react';

declare var twttr: any;

type Props = {
  user: string,
  text?: string,
};

export default class TwitterTweetToButton extends Component<Props> {
  static defaultProps = { text: '' };

  node = null;

  componentDidMount() {
    if (typeof twttr === 'undefined') {
      const s = document.createElement('script');
      s.src = '//platform.twitter.com/widgets.js';
      s.async = true;
      s.id = 'twitter-wjs';
      if (this.node && this.node.parentNode) this.node.parentNode.appendChild(s);
    } else {
      twttr.widgets.load();
    }
  }

  render() {
    const { user, text } = this.props;
    return (
      <div>
        <a
          ref={(node) => (this.node = node)}
          href={`https://twitter.com/intent/tweet?screen_name=${user}&text=${text || ''}`}
          className="twitter-mention-button"
        >
          Tweet to {user}
        </a>
      </div>
    );
  }
}
