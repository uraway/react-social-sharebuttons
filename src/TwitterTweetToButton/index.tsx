import { FC, useEffect } from 'react';

declare let twttr: {
  widgets: {
    load: () => void;
  };
};

export type TwitterTweetToButtonProps = {
  user: string;
  text?: string;
};

const TwitterTweetToButton: FC<TwitterTweetToButtonProps> = (props) => {
  useEffect(() => {
    let script: HTMLScriptElement;

    if (typeof twttr === 'undefined') {
      script = document.createElement('script');
      script.src = '//platform.twitter.com/widgets.js';
      script.async = true;
      script.id = 'twitter-wjs';
      document.getElementsByTagName('head')[0]?.appendChild(script);
    } else {
      twttr.widgets.load();
    }
  }, []);

  const { user, text } = props;
  return (
    <a
      href={`https://twitter.com/intent/tweet?screen_name=${user}&text=${text || ''}`}
      className="twitter-mention-button"
    >
      Tweet to {user}
    </a>
  );
};

export default TwitterTweetToButton;
