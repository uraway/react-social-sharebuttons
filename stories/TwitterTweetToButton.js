import React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs, text } from '@storybook/addon-knobs';
import { TwitterTweetToButton } from '../dist/index.esm.js';

export default storiesOf('TwitterTweetToButton', module)
  .addDecorator(withKnobs)
  .add('without configure', () => <TwitterTweetToButton user={text('user', 'uraway_')} text={text('text', '')} />);
