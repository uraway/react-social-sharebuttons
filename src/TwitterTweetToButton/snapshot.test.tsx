import renderer from 'react-test-renderer';
import TwitterTweetToButton from '.';

it('renders correctly with required props', () => {
  const tree = renderer.create(<TwitterTweetToButton user="user" />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly with other props', () => {
  const tree = renderer.create(<TwitterTweetToButton text="text" user="user" />).toJSON();
  expect(tree).toMatchSnapshot();
});
