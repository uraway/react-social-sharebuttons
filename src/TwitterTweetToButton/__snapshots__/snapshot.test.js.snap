// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with other props 1`] = `
<div>
  <a
    className="twitter-mention-button"
    href="https://twitter.com/intent/tweet?screen_name=user&text=text"
  >
    Tweet to 
    user
  </a>
</div>
`;

exports[`renders correctly with required props 1`] = `
<div>
  <a
    className="twitter-mention-button"
    href="https://twitter.com/intent/tweet?screen_name=user&text="
  >
    Tweet to 
    user
  </a>
</div>
`;
